src/main/webapp/script.js

```diff
@@ -12,42 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Set the link for the sign-in page.
+// Set the sign-in link for the sign-in page.
 function getSignInLink() {
   fetch('/sign-in').then(response => response.json()).then(info => {
-    const linkEl = document.getElementById('link');
-    if(info.status) {
-      linkEl.innerHTML = '<a href=\"' + info.url + '\">Sign out</a>.';
-    } else {
-      linkEl.innerHTML = '<a href=\"' + info.url + '\">Sign in</a>.';
-    }
+    const linkEl = document.getElementById('sign-in-button');
+    linkEl.href = info.url;
   });
 }
 
+// Show sign-in fail message if appropriate.
+function signInFailMessage() {
+  var url = window.location.href;
+  var status = url.split('=')[1];
+
+  if(status === 'fail') {
+    document.getElementById('fail-message').classList.remove('d-none');
+    document.getElementById('normal-title').classList.add('d-none');
+    document.getElementById('normal-message').classList.add('d-none');
+  }
+}
+
 // Set the sign-up link for the sign-up page.
 function getSignUpLink() {
   fetch('/sign-up').then(response => response.text()).then(link => {
-    const linkEl = document.getElementById('sign-up-link');
-    linkEl.innerHTML = 'Sign up <a href=\"' + link + '\">here</a>.';
+    const linkEl = document.getElementById('sign-up-button');
+    linkEl.href = link;
   });
 }
 
 // Get a specific user's data to populate their profile page.
-function getUserData() {
+function getProfilePageData() {
   var url = window.location.href;
   var key = url.split('?')[1];
 
   fetch('/user?' + key).then(response => response.json()).then(userInfo => {
-    document.getElementById('info').innerHTML = 'Email: ' + userInfo.email + '<br> Name: ' + userInfo.username + '<br> Location: ' + userInfo.location + '<br> Bio: ' + userInfo.bio;
-    document.getElementById('profile-pic').src = userInfo.profilePicUrl;
+    document.getElementById('profile-picture').src = userInfo.profilePictureUrl;
+    document.getElementById('username').innerHTML = userInfo.username;
+    document.getElementById('location').innerHTML = userInfo.location;
+    document.getElementById('bio').innerHTML = userInfo.bio;
 
     document.getElementById('username-input').innerHTML = userInfo.username;
     document.getElementById('location-input').innerHTML = userInfo.location;
     document.getElementById('bio-input').innerHTML = userInfo.bio;
 
-    const buttonEl = document.getElementById('edit-button');
     if(!userInfo.isCurrentUser) {
-      buttonEl.classList.add('hidden');
+      document.getElementById('edit-button').classList.add('d-none');
```

Comment:
You could probably use a better name for this? I think this means setting display to none?

Reply:
I didn't make this CSS class. I used this because it seemed to be the normal way to hide things using Bootstrap.